cmd/podman/create.go

```diff
@@ -117,6 +129,7 @@ func createCmd(c *cli.Context) error {
 	options = append(options, libpod.WithShmSize(createConfig.Resources.ShmSize))
 	options = append(options, libpod.WithGroups(createConfig.GroupAdd))
 	options = append(options, libpod.WithIDMappings(*createConfig.IDMappings))
+	options = append(options, libpod.WithRootFS(createConfig.Rootfs))
```

Comment:
We should only do this if Rootfs is actually set. The `WithRootFS()` option should validate its input to make sure an empty-string rootfs is not passed, and the given rootfs directory actually exists

Reply:
I’d conceptually prefer that as well—but from a quick check quite a few of the calls above silently ignore empty strings, from the critical `WithRootFSFromImage` to the more trivial `WithConmonPidFile` or `WithShmDir`.

(Yeah, Golang not having proper nullable types sucks.)

Reply:
We need to fix `WithRootFSFromImage()` as well at a minimum (make it require non-empty input and only set it if we're using an image rootfs).

The rest should also be hit by a validation pass as well at some point.